ares-proxy/main_test.go

```
@@ -0,0 +1,230 @@
+func newTestServer(broker *Broker) *httptest.Server {
+	mux := http.NewServeMux()
+	registerRoutes(mux, broker)
+	return httptest.NewServer(mux)
+}
+
+func pollUntilRequest(t *testing.T, client *http.Client, serverURL string) PendingRequest {
+	t.Helper()
+
+	deadline := time.Now().Add(1 * time.Second)
+	for time.Now().Before(deadline) {
+		resp, err := client.Get(serverURL + "/poll")
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)